# QCTO - Workplace Module

### Project Title: Please Insert your Project Title Here
#### Done By: Name and Surname

© ExploreAI 2024

---

## Table of Contents

<a href=#BC> Background Context</a>

<a href=#one>1. Importing Packages</a>

<a href=#two>2. Data Collection and Description</a>

<a href=#three>3. Loading Data </a>

<a href=#four>4. Data Cleaning and Filtering</a>

<a href=#five>5. Exploratory Data Analysis (EDA)</a>

<a href=#six>6. Modeling </a>

<a href=#seven>7. Evaluation and Validation</a>

<a href=#eight>8. Final Model</a>

<a href=#nine>9. Conclusion and Future Work</a>

<a href=#ten>10. References</a>

---
 <a id="BC"></a>
## **Background Context**
<a href=#cont>Back to Table of Contents</a>

* **Purpose:** Introduce the project, outline its goals, and explain its significance.
* **Details:** Include information about the problem domain, the specific questions or challenges the project aims to address, and any relevant background information that sets the stage for the work.
---

---
<a href=#one></a>
## **Importing Packages**
<a href=#cont>Back to Table of Contents</a>

* **Purpose:** Set up the Python environment with necessary libraries and tools.
* **Details:** List and import all the Python packages that will be used throughout the project such as Pandas for data manipulation, Matplotlib/Seaborn for visualization, scikit-learn for modeling, etc.
---

In [1]:
# Installing own package to check for nulls and duplicates

!pip install git+https://github.com/KGEbrahim/Workplace-project.git@cleaning-branch

  Cloning https://github.com/KGEbrahim/Workplace-project.git (to revision cleaning-branch) to c:\users\kgerr\appdata\local\temp\pip-req-build-s0lw4gmu
  Resolved https://github.com/KGEbrahim/Workplace-project.git to commit 4c03df3af6b4dfc79f76e8168cde36c137fd1ccc
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'


  Running command git clone --filter=blob:none --quiet https://github.com/KGEbrahim/Workplace-project.git 'C:\Users\kgerr\AppData\Local\Temp\pip-req-build-s0lw4gmu'
  f () { sleep 1; echo username=git store: -c: line 2: syntax error: unexpected end of file
  f () { sleep 1; echo username=git store: -c: line 2: syntax error: unexpected end of file
  Running command git checkout -b cleaning-branch --track origin/cleaning-branch
  f () { sleep 1; echo username=git store: -c: line 2: syntax error: unexpected end of file
  branch 'cleaning-branch' set up to track 'origin/cleaning-branch'.
  Switched to a new branch 'cleaning-branch'


In [1]:
# Packages for data cleaning and EDA
import pandas as pd
import numpy as np
import re
# Importing own package
from Packages.Packages import cleaning as cle

# Packages for visualization
import matplotlib.pyplot as plt
import seaborn as sns

# To use statistical tools
from scipy.stats import stats

---
<a href=#three></a>
## **Loading Data**
<a href=#cont>Back to Table of Contents</a>

* **Purpose:** Load the data into the notebook for manipulation and analysis.
* **Details:** Show the code used to load the data and display the first few rows to give a sense of what the raw data looks like.
---

In [2]:
# Importing dataset using pandas and confirming source

df = pd.read_csv(r'C:\Users\kgerr\OneDrive\Documents\ExploreAI\Workplace local\Dataset\yield_df.csv')

# From observing the columns a few code cells below I picked up that the unnamed column was using as an index but this is not needed and will dropped
df.drop(columns=['Unnamed: 0'], inplace=True)

# Creating a copy of the dataframe as a safey measure when modifying the dataframe
df_copy = df
df_copy.head()

Area         Item  Year  hg/ha_yield  average_rain_fall_mm_per_year  \
0  Albania        Maize  1990        36613                         1485.0   
1  Albania     Potatoes  1990        66667                         1485.0   
2  Albania  Rice, paddy  1990        23333                         1485.0   
3  Albania      Sorghum  1990        12500                         1485.0   
4  Albania     Soybeans  1990         7000                         1485.0   

   pesticides_tonnes  avg_temp  
0              121.0     16.37  
1              121.0     16.37  
2              121.0     16.37  
3              121.0     16.37  
4              121.0     16.37

---
<a href=#two></a>
## **Data Collection and Description**
<a href=#cont>Back to Table of Contents</a>

* **Purpose:** Describe how the data was collected and provide an overview of its characteristics.
* **Details:** Mention sources of the data, the methods used for collection (e.g., APIs, web scraping, datasets from repositories), and a general description of the dataset including size, scope, and types of data available (e.g., numerical, categorical).
---

In [11]:
'''
The yield dataset was sourced from Kaggle and represents various features of agricultural production and environmental factors across different countries, this was loaded into the workbook using pandas.

It appears to be cleaned already for creating statistical models.
The dataset contains float, int and object data types, 6 columns including an unnamed column and 28424 rows.
'''

df_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28242 entries, 0 to 28241
Data columns (total 7 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   Area                           28242 non-null  object 
 1   Item                           28242 non-null  object 
 2   Year                           28242 non-null  int64  
 3   hg/ha_yield                    28242 non-null  int64  
 4   average_rain_fall_mm_per_year  28242 non-null  float64
 5   pesticides_tonnes              28242 non-null  float64
 6   avg_temp                       28242 non-null  float64
dtypes: float64(3), int64(2), object(2)
memory usage: 1.5+ MB


In [21]:
df_copy.head()
# It seems that the unnamed column is meant as an index column, I could drop that column seeing that I've already got the standard index to use, this will be done at the initial loading.
# Previously: Unnamed: 0	Area	Item	Year	hg/ha_yield	average_rain_fall_mm_per_year	pesticides_tonnes	avg_temp

Area         Item  Year  hg/ha_yield  average_rain_fall_mm_per_year  \
0  Albania        Maize  1990        36613                         1485.0   
1  Albania     Potatoes  1990        66667                         1485.0   
2  Albania  Rice, paddy  1990        23333                         1485.0   
3  Albania      Sorghum  1990        12500                         1485.0   
4  Albania     Soybeans  1990         7000                         1485.0   

   pesticides_tonnes  avg_temp  
0              121.0     16.37  
1              121.0     16.37  
2              121.0     16.37  
3              121.0     16.37  
4              121.0     16.37

In [13]:
# Basic summary check of the numerical variables

df_copy.describe()

Year    hg/ha_yield  average_rain_fall_mm_per_year  \
count  28242.000000   28242.000000                    28242.00000   
mean    2001.544296   77053.332094                     1149.05598   
std        7.051905   84956.612897                      709.81215   
min     1990.000000      50.000000                       51.00000   
25%     1995.000000   19919.250000                      593.00000   
50%     2001.000000   38295.000000                     1083.00000   
75%     2008.000000  104676.750000                     1668.00000   
max     2013.000000  501412.000000                     3240.00000   

       pesticides_tonnes      avg_temp  
count       28242.000000  28242.000000  
mean        37076.909344     20.542627  
std         59958.784665      6.312051  
min             0.040000      1.300000  
25%          1702.000000     16.702500  
50%         17529.440000     21.510000  
75%         48687.880000     26.000000  
max        367778.000000     30.650000

**Findings**: 
- The dataset seems to contain data from 1990 to 2013.
- The hectograms per hectare (hg/ha) column would require further understanding of the measurement used.
- The rest of the variables are straightforward, grouping of these variables with aggregations and sorting them should give us more insights.

---
<a href=#four></a>
## **Data Cleaning and Filtering**
<a href=#cont>Back to Table of Contents</a>

* **Purpose:** Prepare the data for analysis by cleaning and filtering.
* **Details:** Include steps for handling missing values, removing outliers, correcting errors, and possibly reducing the data (filtering based on certain criteria or features).
---

In [55]:
# Checking for nulls and duplicates

cle.findnulls_duplicates(df_copy)

Number of row duplicates: 2310


null_rows
Area                                   0
Item                                   0
Year                                   0
hg/ha_yield                            0
average_rain_fall_mm_per_year          0
pesticides_tonnes                      0
avg_temp                               0


The dataset appears to be clean from nulls however, 2310 duplicate rows were picked up and will be dropped.

In [5]:
# Assigning non-duplicate values back to df_copy

df_copy = cle.duplicates_removed(df_copy)
df_copy

Area            Item  Year  hg/ha_yield  \
0       Albania           Maize  1990        36613   
1       Albania        Potatoes  1990        66667   
2       Albania     Rice, paddy  1990        23333   
3       Albania         Sorghum  1990        12500   
4       Albania        Soybeans  1990         7000   
...         ...             ...   ...          ...   
25927  Zimbabwe     Rice, paddy  2013        22581   
25928  Zimbabwe         Sorghum  2013         3066   
25929  Zimbabwe        Soybeans  2013        13142   
25930  Zimbabwe  Sweet potatoes  2013        22222   
25931  Zimbabwe           Wheat  2013        22888   

       average_rain_fall_mm_per_year  pesticides_tonnes  avg_temp  
0                             1485.0             121.00     16.37  
1                             1485.0             121.00     16.37  
2                             1485.0             121.00     16.37  
3                             1485.0             121.00     16.37  
4                             1485.0             121.00     16.37  
...                              ...                ...       ...  
25927                          657.0            2550.07     19.76  
25928                          657.0            2550.07     19.76  
25929                          657.0            2550.07     19.76  
25930                          657.0            2550.07     19.76  
25931                          657.0            2550.07     19.76  

[25932 rows x 7 columns]

---
<a href=#five></a>
## **Exploratory Data Analysis (EDA)**
<a href=#cont>Back to Table of Contents</a>

* **Purpose:** Explore and visualize the data to uncover patterns, trends, and relationships.
* **Details:** Use statistics and visualizations to explore the data. This may include histograms, box plots, scatter plots, and correlation matrices. Discuss any significant findings.
---


In [6]:
# Exploring the data


In [5]:
# Exploring the median estimatedsalary


---
<a href=#six></a>
## **Modeling**
<a href=#cont>Back to Table of Contents</a>

* **Purpose:** Develop and train predictive or statistical models.
* **Details:** Describe the choice of models, feature selection and engineering processes, and show how the models are trained. Include code for setting up the models and explanations of the model parameters.
---


In [ ]:
#Please use code cells to code in and do not forget to comment your code.

---
<a href=#seven></a>
## **Evaluation and Validation**
<a href=#cont>Back to Table of Contents</a>

* **Purpose:** Evaluate and validate the effectiveness and accuracy of the models.
* **Details:** Present metrics used to evaluate the models, such as accuracy, precision, recall, F1-score, etc. Discuss validation techniques employed, such as cross-validation or train/test split.
---

In [ ]:
#Please use code cells to code in and do not forget to comment your code.

---
<a href=#eight></a>
## **Final Model**
<a href=#cont>Back to Table of Contents</a>

* **Purpose:** Present the final model and its performance.
* **Details:** Highlight the best-performing model and discuss its configuration, performance, and why it was chosen over others.
---


In [ ]:
#Please use code cells to code in and do not forget to comment your code.

---
<a href=#nine></a>
## **Conclusion and Future Work**
<a href=#cont>Back to Table of Contents</a>

* **Purpose:** Summarize the findings and discuss future directions.
* **Details:** Conclude with a summary of the results, insights gained, limitations of the study, and suggestions for future projects or improvements in methodology or data collection.
---


In [ ]:
#Please use code cells to code in and do not forget to comment your code.

---
<a href=#ten></a>
## **References**
<a href=#cont>Back to Table of Contents</a>

* **Purpose:** Provide citations and sources of external content.
* **Details:** List all the references and sources consulted during the project, including data sources, research papers, and documentation for tools and libraries used.
---

In [ ]:
#Please use code cells to code in and do not forget to comment your code.

## Additional Sections to Consider

* ### Appendix: 
For any additional code, detailed tables, or extended data visualizations that are supplementary to the main content.

* ### Contributors: 
If this is a group project, list the contributors and their roles or contributions to the project.
